src/ServiceControl/MessageFailures/FailedMessage.cs

```diff
@@ -26,6 +26,11 @@ public static string MakeDocumentId(string messageUniqueId)
             return $"{CollectionName}/{messageUniqueId}";
         }
 
+        public static string GetMessageIdFromDocumentId(string failedMessageDocumentId)
+        {
+            return failedMessageDocumentId.Substring(CollectionName.Length + 1);
```

Comment:
If, for whatever reason, the strings are not what we expect, we will throw an exception and stop the whole cleanup process here. Is it worth putting some defensive checks around this?

Reply:
I think for perf reasons it would be fair to assume it is only used when the collection name is a guaranteed match. Any defensive check has associated runtime costs